### ISSUE TITLE
PNG with long iTXt chunk not detected as PNG image

### ISSUE THREAD

**Issue by lenartbezek** [sindresorhus/file-type#320]
Some PNG images have long `iTXt` chunks so the parser never reaches `IDAT` chunk.

When printing chunks after the header, I get something like this:
```
{ length: 13, type: 'IHDR' }
{ length: 9, type: 'pHYs' }
{ length: 25204, type: 'iTXt' }
```

A solution perhaps would be to add a final return after [this line](https://github.com/sindresorhus/file-type/blob/master/core.js#L887) in case the PNG header matches, but no `IDAT` chunk is found in first 4100 bytes. Or perhaps an extra switch case for `ITXt` chunk?

Example image with `ITXt` chunk: https://puu.sh/F4fqT.png

**Comment by Borewit** [comment]
> A solution perhaps would be to add a final return after this line in case the PNG header matches

Maybe a reasonable fallback indeed.

> , but no IDAT chunk is found in first 4100 bytes. Or perhaps an extra switch case for ITXt chunk?

The 4100 sample limit is no longer in file-type (see #248, #319), except in `stream()`.

Which function are you using?

Can you use [`fromFile(filePath)`](https://github.com/sindresorhus/file-type#filetypefromfilefilepath) instead?

**Comment by Borewit** [comment]
I merged your (@lenartbezek) changes to #320, on top of PR #319 to get rid of the 4k side effects.

**Comment by lenartbezek** [comment]
> The 4100 sample limit is no longer in file-type (see #248, #319), except in stream(). 
Which function are you using?

I am using `stream()` when uploading a file and piping it directly into destination. We're not using `fromFile` or `fromBuffer` to avoid saving the entire file locally before uploading it to storage.
